scripts/check_golang_version.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+# shellcheck disable=2072
+
+set -eo pipefail
+
+read -ra GOLANG_VERSIONS <<< "$(./scripts/get_golang_version.sh all)"
+BUILD_VERSION="${GOLANG_VERSIONS[0]}"
+MIN_VERSION="${GOLANG_VERSIONS[1]}"
+GO_MOD_SUPPORT="${GOLANG_VERSIONS[2]}"
+
+# Get the field "go1.1.1" and then remove the "go" prefix.
+SYSTEM_GOLANG_VERSION=$(go version | awk '{ gsub(/go/, "", $3); print $3 }')
+
+# https://golang.org/doc/go1.11#modules
+if [[ "${SYSTEM_GOLANG_VERSION}" < "$GO_MOD_SUPPORT" ]]
+then
+    echo "[$0] ERROR: The version of go on the system (${SYSTEM_GOLANG_VERSION}) is too old and does not support go modules. Please update to at least ${MIN_VERSION}"
+    exit 1
+fi
+
+if [ "$1" == dev ]
+then
+    if [[ "${SYSTEM_GOLANG_VERSION}" < "${MIN_VERSION}" ]]
+    then
+        echo "[$0] WARNING: The version of go on the system (${SYSTEM_GOLANG_VERSION}) is below the recommended version (${MIN_VERSION}) and therefore may not build correctly."
+        echo "[$0] Please update to at least ${MIN_VERSION}"
+    fi
+else
+    # Check to make sure that it matches what is specified in go.mod
+    GOMOD_VERSION=$(go mod edit -json | jq -r ".Go")
+    TRIMMED_BUILD_VERSION=$(awk -F. '{ print $1 "." $2 }' <<< "${BUILD_VERSION}")
+
+    if ! [[ "${BUILD_VERSION}" == "${GOMOD_VERSION}" || "${TRIMMED_BUILD_VERSION}" == "${GOMOD_VERSION}" ]]
```

Comment:
Pre-existing code, I know, but why not just always compare to $TRIMMED_BUILD_VERSION here?

Reply:
In case the pinned version in `go.mod` has a patch version.

Reply:
I don't think that's allowed. That's how we had it initially to pin the version, but it didn't work, which is why we stick with the minor version.

Reply:
I'm going to push a commit that compares the strings.  I had posed this idea when this code was initially merged, and I think it's a succinct and safe way to verify build version.

For example, as long as we compare either string to the beginning of the other, we shouldn't get any false positives.

```
[[ 1.14 =~ ^1.14 ]] # true
[[ 1.14.7 =~ ^1.14 ]] # true
[[ 1.14.7 =~ ^14.7 ]] # false
```